### ISSUE TITLE
Parameter referencing in PathItemObject.parameters.

### ISSUE THREAD

**Issue by mopaul** [mission-liao/pyswagger#105]
My swagger definition defines a bunch of query parameters in the root.parameters  which I then reference from the **pathItem.parameters** section. When I try to access the operation parameters using app.op.parameters, it looks like only the 1st parameter referenced in pathItem.parameters is associated with the path's Operation object. 

If I understand the swagger definition, pathItems.parameters are applicable to all the operations in the path.

Am I missing something?

Referencing a parameter from **operation.parameters** works as expected.

**Comment by mission-liao** [comment]
It should work when each $ref points to one parameter definition under root.parameters. I just add a test case for it, could you check if this is what you expected?
[test-spec](https://github.com/mission-liao/pyswagger/blob/master/pyswagger/tests/data/v2_0/resolve/other/swagger.json)
[test-case](https://github.com/mission-liao/pyswagger/blob/master/pyswagger/tests/v2_0/test_resolve.py#L54)

If this is not what you described, could you help to provide a sample swagger.json/yaml and code?

**Comment by mopaul** [comment]
I meant referencing a parameter in the path.parameter section. Something like this. 

```json
"paths":{
      "/a":{
         "get":{
            "parameters":[
               {
                  "name":"p2",
                  "in":"body",
                  "schema":{
                     "$ref":"#/definitions/d1"
                  }
               }
            ],
            "responses":{
               "default":{
                  "$ref":"#/responses/r1"
               }
            }
         },
         "parameters": [
            {
               "$ref":"#/parameters/p1"
            },
            {
               "$ref":"#/parameters/p2"
            }
         ]
      }
   }
```

The parameters p1 and p2 are being referenced from the path.parameters and not operation.parameters. I can only see p2 and p1_d associated with the operation but (if I understand the swagger spec correctly), p2, p1_d and p2_d are all valid parameters. 

Let me know if you want me to send a PR with a test case.

**Comment by mission-liao** [comment]
The case you described is new to me, and yes, it's spec compliant. Any PR with test case is welcome, thanks.

**Comment by mopaul** [comment]
[PR 106](https://github.com/mission-liao/pyswagger/pull/106) has the test case.

**Comment by mission-liao** [comment]
OK, thanks. I'll work on this part tomorrow evening.

**Comment by mission-liao** [comment]
this bug is fixed in v0.8.26, please feel free to reopen it when it's still failed.

**Comment by mopaul** [comment]
Fix verified. Thanks for your quick responses.
